### PR TITLE
Fix various type issues and iterators

### PR DESCRIPTION
**Pull request by local-minimum**
Unfortunately this reverts the fix that created the `analysis` folder because it was not done correctly before. the output folder field may not include `/`

### CODE REVIEW

File: scanomatic/server/analysis_effector.py [hunk 1]
```diff
@@ -108,6 +110,10 @@ def total(self) -> int:
             return self._first_pass_results.total_number_of_images
         return -1
 
+    @property
+    def ready_to_start(self) -> bool:
```
Comment (skymandr) [inline scanomatic/server/analysis_effector.py:114]: Du använder inte denna..?

File: scanomatic/server/compile_effector.py [hunk 2]
```diff
@@ -180,9 +180,9 @@ def __next__(self):
             is COMPILE_ACTION.InitiateAndSpawnAnalysis
         ):
             self._spawn_analysis()
-            self.enact_stop()
+            self.enact_stop()  # Raises StopIteration
```
Comment (skymandr) [inline scanomatic/server/compile_effector.py:183]: Varför inte bara `raise`?

**Review by skymandr — APPROVED** [review]
Looks good, a few minor questions.

**Comment by skymandr** [comment]
> Unfortunately this reverts the fix that created the `analysis` folder because it was not done correctly before. the output folder field may not include `/`

Reopen that issue?

File: scanomatic/server/analysis_effector.py (hunk 1 above)
Comment (local-minimum) [reply scanomatic/server/analysis_effector.py:114]: in the new test. wanted to have a separate test to show setup is ready

File: scanomatic/server/compile_effector.py (hunk 2 above)
Comment (local-minimum) [reply scanomatic/server/compile_effector.py:183]: yes why not... I'll move the raise here

File: scanomatic/server/analysis_effector.py (hunk 1 above)
Comment (skymandr) [reply scanomatic/server/analysis_effector.py:114]: ah, right. saw the test, but was wondering if it was yagni since you don't use it operationally.

**Comment by local-minimum** [comment]
> > Unfortunately this reverts the fix that created the `analysis` folder because it was not done correctly before. the output folder field may not include `/`
> 
> Reopen that issue?

Not needed, it will have to be solved in the integration tests I believe.

**Pull request merged**